connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -491,6 +490,10 @@ private void closePartitions() {
     private class HandleRebalance implements ConsumerRebalanceListener {
         @Override
         public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
+            // Increase the commit sequence number so that asynchronous commits that are not complete
+            // won't overwrite what we do here
+            commitSeqno += 1;
+            committing = false;
```

Comment:
Ah, maybe my test case doesn't correctly reflect the normal consumer behavior. The consumer will call `onPartitionsRevoked(...)` before it calls `onPartitionsAssigned(...)`?